ui/perfherder/alerts/AlertTable.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+import { Container, Form, FormGroup, Label, Input, Table } from 'reactstrap';
+
+import perf from '../../js/perf';
+
+import AlertHeader from './AlertHeader';
+import StatusDropdown from './StatusDropdown';
+
+// TODO remove $stateParams and $state after switching to react router
+export class AlertTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alertSummary: this.props.alertSummary,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { alertSummary } = this.props;
+    if (prevProps.alertSummary !== alertSummary) {
+      // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
Would this be more appropriate in ``getDerivedStateFromProps``?  I don't think you get access to ``prevProps`` in that function, though.  So perhaps not.

Reply:
I've looked into this code again and it doesn't seem to be behaving as expected (might be a side effect of doing partial updates with angular) so I've removed it. I'll attempt to keep a single "source of truth" for this object during the rest of the conversion so that there won't be a need for either of those life cycle methods.